### DISCUSSION TITLE
Bluetooth Audio over Long Distance 900m with (2.4 GHz Nrf24l01 E01-ML01DP5 Ebyte)

### DISCUSSION THREAD

**Discussion by MeesMeijer** [pschatzmann/arduino-audio-tools#1472] [Q&A] [answered]
Hey there, 

I'm trying to make a long-distance audio streaming solution for my missing music problem.
This is because I want to listen to music when wakeboarding. I'm aware that RF waves don't go through water, so I will have an antenna on my helmet. 
But first, I'm trying to build a prototype to see if this even works. 

The planned data flow: 
phone -> esp + bluetooth -> RF24lo1 -> RF_WAVES -> RF24l01 -> esp32 +bluetooth -> headphones

I've tried a lot of code and examples from this repo, but nothing works... 

Any suggestions? 

esp32: ESP-WROOM-32 
Headphones: Sony NWWS623
Transmitters: 2.4 GHz Nrf24l01 E01-ML01DP5 Ebyte 20db

**Comment by pschatzmann** [comment, marked as answer]
I guess your bottleneck is the RF24l01. 
As a starting point, try to measure what thruput you get. 
With that you can calculate the max sample rate for stereo 16 bit data that you can use to transmit: It will be much less then your 44100 that you get from your A2DP!

Finally  you can create a test sketch  to confirm your max sample rate and you can experiment with different codecs. 

and last but not least consider the [advice from the Wiki](https://github.com/pschatzmann/arduino-audio-tools/wiki/It's-not-working)
